apiml-security-common/src/main/java/org/zowe/apiml/security/common/content/AbstractSecureContentFilter.java

```diff
@@ -79,6 +79,14 @@ protected void doFilterInternal(@NonNull HttpServletRequest request, @NonNull Ht
                 failureHandler.onAuthenticationFailure(request, response, authenticationException);
             } catch (RuntimeException e) {
                 resourceAccessExceptionHandler.handleException(request, response, e);
+            } finally {
+                Authentication authentication = authenticationToken.get();
+                if (authentication != null) {
+                    Object credentials = authenticationToken.get().getCredentials();
+                    if (credentials instanceof char[]) {
+                        Arrays.fill((char[]) credentials, (char) 0);
```

Comment:
Actually, this is more-less responsibility of Spring security, see org.springframework.security.core.CredentialsContainer#eraseCredentials